### PR TITLE
HA flows Part 9: Add H&S delete operation

### PR DESCRIPTION
**Pull request by niksv**
API delete stub was replaced with HaFlowDeleteFSM. This FSM deallocates resources and pathes for ha-flows but doesn't rempve rules yet

Related To #5061

### CODE REVIEW

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/HaFlowDeleteHubBolt.java [hunk 1]
```diff
@@ -15,93 +15,190 @@
...
+        HaFlowDeleteRequest request = pullValue(input, FIELD_ID_PAYLOAD, HaFlowDeleteRequest.class);
         try {
-            handleHaFlowDelete(key, payload);
-        } catch (FlowProcessingException e) {
-            emitErrorMessage(e.getErrorType(), COMMON_ERROR_MESSAGE, e.getMessage());
-        } catch (Exception e) {
-            emitErrorMessage(ErrorType.INTERNAL_ERROR, COMMON_ERROR_MESSAGE, e.getMessage());
+            service.handleRequest(currentKey, getCommandContext(), request.getHaFlowId());
+        } catch (DuplicateKeyException e) {
+            log.error("Failed to handle a request with key {}. {}", currentKey, e.getMessage());
```
Comment (dmitrii-beliakov) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/HaFlowDeleteHubBolt.java:109]: I'd add the type into the message itself. Also `key` is countable. 
something like: `Failed to handle a request due to the duplicated request key {}` (or what kind of key is it?)

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/create/actions/ResourcesAllocationAction.java [hunk 2]
```diff
@@ -378,12 +378,6 @@ protected void handleError(HaFlowCreateFsm stateMachine, Exception ex, ErrorType
...
     }
 
-    protected HaFlow getHaFlow(String haFlowId) {
-        return haFlowRepository.findById(haFlowId)
-                .orElseThrow(() -> new FlowProcessingException(ErrorType.NOT_FOUND,
-                        format("HA-flow %s not found", haFlowId)));
-    }
-
     private FlowSubType getSubType(int id) {
         //TODO find a better way
```
Comment (dmitrii-beliakov) [inline src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/create/actions/ResourcesAllocationAction.java:382]: Are you going to find a better way? If no, then I'd create a follow-up task.
The logic probably could be moved to enum itself: have a method FlowSubType.of(int)

**Review by dmitrii-beliakov — APPROVED** [review]

**Comment by artHalMirantis** [comment]
RP can be merged, autochecks are enough

File: src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/logger/FlowOperationsDashboardLogger.java [hunk 3]
```diff
@@ -754,4 +756,41 @@ public void onFailedHaFlowCreate(String haFlowId, String failureReason) {
...
+    public void onHaFlowDelete(String haFlowId) {
+        Map<String, String> data = new HashMap<>();
+        data.put(TAG, "ha-flow-delete");
+        data.put(FLOW_ID, haFlowId);
+        data.put(EVENT_TYPE, HA_FLOW_DELETE_EVENT);
+        invokeLogger(Level.INFO, String.format("Delete the ha-flow %s", haFlowId), data);
+    }
+
+    /**
+     * Log a ha-flow-delete-successful event.
```
[the remark this answers is not in the document]
Comment (niksv) [reply src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/logger/FlowOperationsDashboardLogger.java:772]: idea disagree with you :D

Screenshot from 2023-04-01 00-09-31

[the remark this answers is not in the document]
Comment (dmitrii-beliakov) [reply src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/logger/FlowOperationsDashboardLogger.java:772]: yeah, it's tricky and probably based on region as well. Articles used based on sound. Here `ha` sounds like `[cha]` (like haha) and when a word starts with a consonant we use `a`. Like `a high salary` this `high` sounds like `[chai]`. But in words like `hour`, that h is silent and sounds like `[our]` or `[aur]`, in this case it is an hour. To me, HA as an abbreviation sounds like [eitch] or [aitch] and should be used with an. Similar to `HR`: an HR manager `[en eitch ar ...].

File: src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/logger/FlowOperationsDashboardLogger.java [hunk 4]
```diff
@@ -754,4 +756,41 @@ public void onFailedHaFlowCreate(String haFlowId, String failureReason) {
...
+
+    /**
+     * Log a flow-delete-failed event.
+     */
+    public void onFailedHaFlowDelete(String haFlowId, String failureReason) {
+        Map<String, String> data = new HashMap<>();
+        data.put(TAG, "ha-flow-delete-failed");
+        data.put(FLOW_ID, haFlowId);
+        data.put(EVENT_TYPE, HA_FLOW_DELETE_RESULT_EVENT);
+        data.put("delete-result", "failed");
```
[the remark this answers is not in the document]
Comment (niksv) [reply src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/logger/FlowOperationsDashboardLogger.java:791]: Fixed

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/FlowHsTopology.java [hunk 5]
```diff
@@ -709,10 +712,28 @@ private void haFlowUpdateHub(TopologyBuilder topologyBuilder, PersistenceManager
     }
 
     private void haFlowDeleteHub(TopologyBuilder topologyBuilder, PersistenceManager persistenceManager) {
-        HaFlowDeleteHubBolt hubBolt = new HaFlowDeleteHubBolt(persistenceManager);
+        int hubTimeout = (int) TimeUnit.SECONDS.toMillis(topologyConfig.getDeleteHubTimeoutSeconds());
+
+        HaFlowDeleteConfig config = HaFlowDeleteConfig.haFlowDeleteBuilder()
+                .speakerCommandRetriesLimit(topologyConfig.getDeleteSpeakerCommandRetries())
+                .autoAck(true)
+                .timeoutMs(hubTimeout)
```
[the remark this answers is not in the document]
Comment (niksv) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/FlowHsTopology.java:720]: good idea

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/HaFlowDeleteHubBolt.java (hunk 1 above)
Comment (niksv) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/HaFlowDeleteHubBolt.java:109]: fixed

File: src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/create/actions/ResourcesAllocationAction.java (hunk 2 above)
Comment (niksv) [reply src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/create/actions/ResourcesAllocationAction.java:382]: will create a task

**Pull request merged**